docker/Makefile

```diff
@@ -1,39 +1,34 @@
+include ../make-base/utils.mk
 export COMPOSE_PROJECT_NAME=monoetna
-projects := $(shell ls ../*/Makefile | grep -v docker | xargs -n 1 dirname | xargs -n 1 basename)
-compose_ymls     := $(shell ls ../*/docker-compose.yml | grep -v 'docker/docker-compose.yml')
-images := $(shell echo */)
+compose_ymls:=$(call find_projects_containing,docker-compose.yml)
+compose_projects:=$(call map,dirname,$(compose_ymls))
 
+.PHONY: help
 help: ## Display help text
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) /dev/null | \
 		sed 's/^[^:]*://' | sort | \
 		awk -F':.*?## ' '{printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
-.PHONY: help
 .DEFAULT_GOAL := help
 
-docker-compose.yml: $(compose_ymls) ./compose $(wildcard ./*.shared.yml) ./default_compose
-	@ for project in $(projects); do cd ../$$project && make docker-compose.yml; done
+docker-compose.yml: $(compose_ymls) ./build_support/compose $(wildcard ./*.shared.yml)
 	@ echo 'Rebuilding docker-compose.yml from projects $(compose_ymls)'
-	@ ./compose $(compose_ymls) > .tmp
+	@ ./build_support/compose $(compose_ymls) > .tmp
 	@ mv .tmp docker-compose.yml
 
 .PHONY: config-ready
 config-ready: docker-compose.yml ## Prepares the top level docker-compose.yml config.  Run this if you recently changed docker-compose.yml files.
 
-.PHONY: docker-ready
-docker-ready: config-ready build
-
-.PHONY: build
-build: ## Builds all development related docker images
```

Comment:
`build` used to build every single development image together.  Now, dependencies are actualized and this enmasse build is only necessary for a root level `make up`.  If you instead run `make -C janus up` you will only build janus dependencies, rather than build the entire world.